core/templates/domain/statistics/LearnerActionObjectFactory.ts

```diff
@@ -71,45 +76,35 @@ export class LearnerAction {
   providedIn: 'root'
 })
 export class LearnerActionObjectFactory {
-  /**
-   * @property {string} actionType - type of an action
-   * @property {Object.<string, *>} actionCustomizationArgs - customization
-   *   dict for an action
-   * @property {number} [schemaVersion=LEARNER_ACTION_SCHEMA_LATEST_VERSION]
-   *   - schema version of the class instance.
-   * @returns {LearnerAction}
-   */
-  createNew(
-      actionType: string, actionCustomizationArgs: any,
-      schemaVersion: number): LearnerAction {
-    schemaVersion = schemaVersion ||
-      StatisticsDomainConstants.LEARNER_ACTION_SCHEMA_LATEST_VERSION;
+  createFromBackendDict(backendDict: ILearnerActionBackendDict): LearnerAction {
+    return new LearnerAction(
+      backendDict.action_type,
+      backendDict.schema_version,
+      backendDict.action_customization_args);
+  }
+
+  createExplorationStartAction(
+      customizationArgs: IExplorationStartCustomizationArgs): LearnerAction {
```

Comment:
Instead of defining interfaces for all this stuff and passing them in as dict blobs, I think it would be better to pass them as separate args. Would that be possible? I would like to avoid an explosion of custom types, especially when these can be avoided / don't really represent a strongly-cohesive concept.

Ditto below.

Reply:
Hmm I don't really see a way around it since the customization args would need to be placed _somewhere_, and given there are 3 sets of args we'd need 3 sets of types.

With that said, I can cut down the types by inlining the customization args into the backend dict interface, would that be any better?

Reply:
Mm. Would you still need the concept of "customization args" in the frontend for this case?

Reply:
I think so, given that we need to build them up to send it to the backend. We also need the values from the backend to render these properly (generate sentences etc.).